### ISSUE TITLE
Parameter properties information not being capture in ESTree parameter node

### ISSUE THREAD

**Issue by weirdpattern** [eslint/typescript-eslint-parser#143]
**What version of TypeScript are you using?**
2.0.10 and 2.1.5

**What version of `typescript-eslint-parser` are you using?**
1.0.2

**What code were you trying to parse?**

```ts
class Foo {
    constructor(public name : string) {}
}
```

```ts
class Foo {
    constructor(protected name : string) {}
}
```

```ts
class Foo {
    constructor(private name : string) {}
}
```

```ts
class Foo {
    constructor(readonly name : string) {}
}
```

```ts
class Foo {
    constructor(public readonly name : string) {}
}
```

```ts
class Foo {
    constructor(protected readonly name : string) {}
}
```

```ts
class Foo {
    constructor(private readonly name : string) {}
}
```

**What did you expect to happen?**
Parameter properties should be parsed correctly and modifiers should be part of the final ESTree parameter node.

**What happened?**
The modifiers property of the TS Node is being ignored by the AST Converter. The final ESTree contains the parameters, but without the accessibility modifiers and/or readonly modifiers.

Sources: [Parameter properties](http://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties)

**Comment by weirdpattern** [comment]
I'm currently working on this. So far I can think of two possible solutions:
1. Add the `isReadonly` and `accessibility` properties to the current output node.
2. Wrap the current output in a new `TSParameterProperty` node. 

This is the current code
<img width="859" alt="constructor" src="https://cloud.githubusercontent.com/assets/19519411/21952885/908ab318-d9ed-11e6-8604-079eeb568437.png">

As parameter properties can only be declare in constructors, I think this is the best place to do the change.

What do you think @nzakas? Adding properties or new node?

**Comment by soda0289** [comment]
I think it should be a new node, TsParameterProperty makes sense. The ESTree already allows params to contain different kinds of pattern nodes such as: AssigmentPattern, ObjectPattern, RestElement, and so on. A new node would allow for more accurate rules that could align better with Class property rules, which might not make sense with method parameter rules.

**Comment by weirdpattern** [comment]
I agree, I will work on this tomorrow
